evm_loader/ERC20/test/test.py

```diff
@@ -138,383 +44,418 @@ def deployERC20(self, location_hex, location_bin,  mintId, balance_erc20):
+            self.assertLess(data[1], 0xd0)  # less 0xd0 - success
+            value = data[2:]
+            balance = int.from_bytes(value, "little")
+            print('balance_ext:', balance)
+            return balance
+
+    def erc20_mint_id(self, erc20, erc20_code):
         input = bytearray.fromhex("03e132a122")
         trx = Transaction().add(
             TransactionInstruction(program_id=self.loader.loader_id, data=input, keys=
             [
                 AccountMeta(pubkey=erc20, is_signer=False, is_writable=True),
+                AccountMeta(pubkey=erc20_code, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.caller, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.acc.public_key(), is_signer=True, is_writable=False),
                 AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
                 AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
             ]))
 
-        result = http_client.send_transaction(trx, self.acc)
-        result = confirm_transaction(http_client, result["result"])
+        result = send_transaction(client, trx, self.acc)
         messages = result["result"]["meta"]["logMessages"]
-        res = messages[messages.index("Program log: succeed") + 1]
-        if not res.startswith("Program log: "):
-            raise Exception("Invalid program logs: no result")
+        res = messages[-1]
+        if any(search("Program %s failed" % evm_loader_id, m) for m in messages):
+            raise Exception("Invalid program logs: Program %s failed" % evm_loader_id)
         else:
-            return res[13:]
+            return res
 
     def test_erc20(self):
-        mintId = self.createMint()
-        time.sleep(20)
-        print("\ncreate token:", mintId)
-        acc_client = self.createTokenAccount(mintId)
-        print ("create account acc_client:", acc_client)
-        balance_erc20 = self.createTokenAccount(mintId)
-        print ("create account balance_erc20:", balance_erc20)
-
-        deploy_result= self.loader.deployERC20("erc20_ctor_uninit.hex", "erc20.bin",  mintId, balance_erc20)
-        erc20Id = deploy_result["programId"]
-        erc20Id_ether = bytearray.fromhex(deploy_result["ethereum"][2:])
-
-        print ("erc20_id:", erc20Id)
-        print ("erc20_id_ethereum:", erc20Id_ether.hex())
-        time.sleep(20)
-        print("erc20 balance_ext():", self.erc20_balance_ext( erc20Id))
-        print("erc20 mint_id():", self.erc20_mint_id( erc20Id))
-
-        self.changeOwner(balance_erc20, erc20Id)
-        print("balance_erc20 owner changed to {}".format(erc20Id))
+        token = self.createToken()
+        print("token:", token)
+
+        balance_erc20 = self.createTokenAccount(token, self.erc20Id_precalculated)
+        print("balance_erc20:", balance_erc20)
+
+        (erc20Id, erc20Id_ether, erc20_code) = self.loader.deploy_erc20("ERC20.bin"
```

Comment:
ERC20.bin is not exists in the repo

Reply:
ERC20.bin is built from ERC20.sol every CI session

Reply:
CI - it is next task.  Is it right?

Reply:
The test runs in CI with this PR